### PR TITLE
Add workaround for broken LIBRARY_SEARCH_PATHS in local Gutenberg

### PR DESCRIPTION
**Pull request by mokagio**
Similar to @twstokes work in https://github.com/wordpress-mobile/WordPress-iOS/pull/21960. The difference is that this makes the change as part of the CocoaPods install. 

I _think_ this might be more robust. By checking for the path on every `LOCAL_GUTENBERG=true bundle exec pod install`, we would prevent the React Native post install hook (or whichever process went and updated that path in the project file) from undoing any fix we track in the project as a one-off.

I also like that we don't change the current search paths unless we have to. I don't think there's anything wrong in setting them to `$(SDKROOT)/usr/lib/swift $(inherited)`, but since it works for the "standard" setup as only `$(inherited)` I'd prefer to keep it that way.

## Testing

Run `LOCAL_GUTENBERG=true bundle exec pod install` from this branch and observe the project file diff has `LIBRARY_SEARCH_PATHS` correctly.

image

The diff is how we expected it and I saw this configuration work at least once. But I also kept running in random issues such as:

image

or:

```
/Users/gio/Developer/a8c/wpios/DerivedData/WordPress/Build/Products/Debug-iphonesimulator/Jetpack.app
is not a valid path to an executable file.
```

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

**PR submission checklist:**

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

**UI changes testing checklist:** Not a UI PR.

### CODE REVIEW

File: Gutenberg/cocoapods_helpers.rb [hunk 1]
```diff
@@ -126,6 +126,11 @@ def gutenberg_post_install(installer:)
   react_native_post_install(installer, react_native_path)
 end
 
+def gutenberg_post_integrate
+  # If the this workaround runs in the post_install step, the changes it makes get overridden somehow.
+  workaround_broken_search_paths
+end
```
Comment (mokagio) [inline Gutenberg/cocoapods_helpers.rb:132]: This tells us something interesting about the possible culprit.

- Has something changed in the project integration phase in CocoaPods that could be formatting the path in that incorrect way?
- Or, is there something that React Native is capable of running at that time, even though we don't call it explicitly?

File: Gutenberg/cocoapods_helpers.rb [hunk 2]
```diff
@@ -172,3 +177,27 @@ def react_native_version!(gutenberg_path:)
...
   package_json_version.split('.').map(&:to_i)
 end
+
+# A workaround for the issue described at
+# https://github.com/wordpress-mobile/WordPress-iOS/pull/21504#issuecomment-1789466523
+#
+# For some yet-to-discover reason, something in the process installing the pods
+# using local sources messes up the LIBRARY_SEARCH_PATHS.
+def workaround_broken_search_paths
+  project = Xcodeproj::Project.open('WordPress/WordPress.xcodeproj')
```
Comment (mokagio) [inline Gutenberg/cocoapods_helpers.rb:187]: There might be a way to read this setting from CocoaPods, but I don't think we need to worry about being that flexible here, right?

File: Gutenberg/cocoapods_helpers.rb [hunk 3]
```diff
@@ -172,3 +177,27 @@ def react_native_version!(gutenberg_path:)
...
+
+  library_search_paths_key = 'LIBRARY_SEARCH_PATHS'
+  broken_search_paths = '$(SDKROOT)/usr/lib/swift$(inherited)'
```
Comment (mokagio) [inline Gutenberg/cocoapods_helpers.rb:190]: Likewise, this hardcoded string makes the workaround work only in one particular case. But I don't think we need to make it more flexible. Better to find the root cause.

**Comment by mokagio** [comment]
@iangmaia 

image

Yes! Super helpful. I forgot to run `bundle exec pod install` because I was testing my changed with `LOCAL_GUTENBERG=true bundle exec pod install` and discarded the lockfile edits.

🎉

**Comment by dcalhoun** [comment]
> [...] But I also kept running in random issues such as:
> 
> image

If I am not mistaken, I believe this relates to https://github.com/wordpress-mobile/gutenberg-mobile/pull/6061#issuecomment-1680532174 and can be resolved by running `rm -rf build/` before installing Pods via `bundle exec pod install`.

**Comment by iangmaia** [comment]
> @iangmaia
> 
> image
> Yes! Super helpful. I forgot to run `bundle exec pod install` because I was testing my changed with `LOCAL_GUTENBERG=true bundle exec pod install` and discarded the lockfile edits.
> 
> 🎉

Yay, nice! 🎉 
In the case of `Podfile`s, we could consider making Danger to fail, given it is a much more certain check than `Gemfile`s and others.

**Comment by mokagio** [comment]
@iangmaia @twstokes @iangmaia thank you for taking a look gents. How do you feel about reviewing this and trying it out in real life?

**Comment by mokagio** [comment]
@dcalhoun 

> If I am not mistaken, I believe this [the React file not found error] relates to https://github.com/wordpress-mobile/gutenberg-mobile/pull/6061#issuecomment-1680532174 and can be resolved by running `rm -rf build/` before installing Pods via `bundle exec pod install`.

When running the Pods command from the WP repo, is the build folder to delete still in `gutenberg-mobile` or does it somehow ends up in WordPress? Would it be helpful if the process did the cleanup for us?

**Comment by dcalhoun** [comment]
> When running the Pods command from the WP repo, is the build folder to delete still in `gutenberg-mobile` or does it somehow ends up in WordPress? Would it be helpful if the process did the cleanup for us?

There is definitely a `build` folder in `WordPress-iOS`. I do not believe the `build` folder within `gutenberg-mobile` is relevant for the specific context of using `LOCAL_GUTENBERG=true bundle exec pod install`. Automating the removal of `WordPress-iOS`' `build` folder during `pod install` would likely avoid a lot of cryptic errors.

**Review by dcalhoun — COMMENTED** [review]
I was able to use these changes to successfully build the app connected to a Metro server running in the `gutenberg-mobile` repository. 🚀

**Comment by iangmaia** [comment]
Also confirming I successfully managed to:

1. Remove any existing `Pods/`, `build/`, `vendor/`, `DerivedData` and so on
2. Run `LOCAL_GUTENBERG=true bundle exec pod install`
3. Confirm the diff in the update `project.pbxproj` correctly didn't break `LIBRARY_SEARCH_PATHS `:
```diff
-				LIBRARY_SEARCH_PATHS = "$(inherited)";
+				LIBRARY_SEARCH_PATHS = (
+					"$(SDKROOT)/usr/lib/swift",
+					"$(inherited)",
+				);
```
4. Compile and run the project.


For the record, as expected, this got printed in the console:
```
Integrating client project
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for WordPress target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for WordPress target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for WordPress target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for WordPress target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for Jetpack target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for Jetpack target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for Jetpack target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
[Gutenberg] Post-processed LIBRARY_SEARCH_PATHS for Jetpack target to fix incorrect '$(SDKROOT)/usr/lib/swift$(inherited)' value.
```

**Review by iangmaia — APPROVED** [review]
👍

**Review by twstokes — APPROVED** [review]
Worked for me @mokagio! Thanks for the fix. 🚀

**Comment by mokagio** [comment]
Thanks folks for verifying the fix.

@dcalhoun :

>  Automating the removal of WordPress-iOS' build folder during pod install would likely avoid a lot of cryptic errors.

Alrighty. I'll merge this PR as is to unblock, then look into automating this bit, too.

**Pull request merged**